### ISSUE TITLE
FreeBSD support - now that Electron is available

### ISSUE THREAD

**Issue by quackerd** [microsoft/vscode#74857]
Hello Devs,

Electron 4 was added to FreeBSD ports and pkg on May 28. Is it possible now to port VSCode to FreeBSD? It would be very much appreciated as there are very few decent GUI code editors on FreeBSD right now.

**Comment by egamma** [comment]
see also https://github.com/microsoft/vscode/issues/1359

**Comment by lapo-luchini** [comment]
The author of the FreeBSD Port of Electron4 is also working on a [FreeBSD Port or VSCode](https://github.com/tagattie/FreeBSD-VSCode).

It is not yet “clean enough” to be committed to the FreeBSD repository (and thus generate an official package), but in the [release section](https://github.com/tagattie/FreeBSD-VSCode/releases) there is a working 1.35.0 release already.

**Comment by quackerd** [comment]
@lapo-luchini Thanks for the information! Does that mean VSCode will official support FreeBSD when the port gets to a clean enough state?

**Comment by lapo-luchini** [comment]
It means FreeBSD will support VSCode (as in: maintaining a Makefile and a set of patches that make it build in the server that build official packages) and it will be easy to install as a package, but whether VSCode wants to support FreeBSD (as in: upstreaming patches that are needed for it to build, listing it as supported on the official website and so on) is up to Microsoft I guess.

**Comment by dyeske** [comment]
I sniffed the current implementation and it appears to download a bunch of binaries from azure.  These binaries look like they contain some Linux nodejs items.  FreeBSD binaries could be built that contain the same functionality.  I just opened https://github.com/microsoft/vscode/issues/83169.

**Comment by quackerd** [comment]
For updates on FreeBSD Electron4 see  [https://github.com/electron/electron/issues/3797](url)

**Comment by PieterSoftware** [comment]
VSCode FreeBSD port is now available: https://twitter.com/FreeBSDHelp/status/1209658237131214848?s=20

**Comment by davidchisnall** [comment]
It would be great to see [the patches](https://svnweb.freebsd.org/ports/head/editors/vscode/files/) that it uses reviewed for upstreaming and to understand what the blockers are to this becoming an officially supported platform.

**Comment by mike-pt** [comment]
There's been tons of votes for getting vscode remote ssh support; which needs this one to happen first!

I wonder if it would be easier to get the maintainer to submit the patches in a PR here. Problem is those are patches to electron too right? not just vscode

**Comment by davidchisnall** [comment]
Remote SSH support does not need this.  Remote SSH just requires the remote agent (which is not open source) to be built for FreeBSD.  Or, ideally, to either be released as open source or have the protocol documented.  There are a load of environments where I'd love for the SSH agent to run but where I definitely wouldn't want to run the IDE.  While I'd love to have a supported build of VS Code for FreeBSD, being able to use it to edit and build things that are on FreeBSD VMs or remote systems would be more valuable to me (maybe less valuable to others) and should have orthogonal requirements.

**Comment by mike-pt** [comment]
I though so to, but not according to this: https://github.com/microsoft/vscode-remote-release/issues/1730#issuecomment-603319916

**Comment by nhooyr** [comment]
For anyone who wants a remote development environment, code-server works on FreeBSD! It's VS Code but running in the browser instead of electron.

https://github.com/cdr/code-server#quick-install

**Comment by grahamperrin** [comment]
@tagattie with [editors/vscode](https://www.freshports.org/editors/vscode/) at `1.46.1`, please: when might we have an update?

https://code.visualstudio.com/updates/ `1.52.1` at the time of writing. 

Thank you

**Comment by davidchisnall** [comment]
It looks as if all of the work to make [1.56 run on FreeBSD](https://www.freshports.org/editors/vscode/) has been done by the community.  Please could someone from the VS Code team document the process that's required to make this into an officially supported platform?

**Comment by olegsidorkin** [comment]
Will patches that bring FreeBSD support be accepted (e.g. adding 'freebsd' os case to node-pty?), or it doesn't worth trying?

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by dyeske** [comment]
So there is an sshfs plugin for vscode.  I've been using it for the last couple of years in a work environment.  It's definitely usable once you get stuff set up.  With some effort I was able to get php and some other languages to have the outline feature work when using vscode with the sshfs plug-in.  On windows pageant works great for this once you set it up.  I ended up needing to compile pageant fresh from git though.  This wasn't that bad once I got cmake set up.  If you plan on connecting as root to a sshfs target running FreeBSD you will want to create a bunch of directories in forward slash so vscode can limp with the workspace.  The main one will be /.vscode but you might need others.  You might also need to touch some files ymmv.
